connector_prestashop/models/account_invoice/importer.py

```diff
@@ -0,0 +1,234 @@
+
+    def _open_refund(self, binding):
+        invoice = binding.odoo_id
+        if invoice.amount_total == float(self.prestashop_record['amount']):
+            invoice.action_invoice_open()
+        else:
+            self.backend_record.add_checkpoint(
+                invoice,
+                message=_('The refund for order %s has a different amount '
+                          'in PrestaShop and in Odoo.') % invoice.origin
+            )
+
+    def _after_import(self, binding):
+        super(RefundImporter, self)._after_import(binding)
+        self._open_refund(binding)
+
+
+class RefundMapper(Component):
+    _name = 'prestashop.refund.mapper'
+    _inherit = 'prestashop.import.mapper'
+    _apply_on = 'prestashop.refund'
+
+    direct = [
+        ('id', 'name'),
+        ('date_add', 'date_invoice'),
+    ]
+
+    @mapping
+    def journal(self, record):
+        journal = self.backend_record.refund_journal_id
+        if not journal:
+            raise MappingError(
+                _('The refund journal must be configured on '
+                  'the PrestaShop Backend.')
+            )
+        return {'journal_id': journal.id}
+
+    def _get_order(self, record):
+        binder = self.binder_for('prestashop.sale.order')
+        return binder.to_internal(record['id_order'])
+
+    @mapping
+    def from_sale_order(self, record):
+        sale_order = self._get_order(record)
+        fiscal_position = None
+        if sale_order.fiscal_position_id:
+            fiscal_position = sale_order.fiscal_position_id.id
+        if sale_order:
+            partner = sale_order.partner_invoice_id
+        else:
+            binder = self.binder_for('prestashop.res.partner')
+            partner = binder.to_internal(record['id_customer'], unwrap=True)
+        return {
+            'origin': sale_order['name'],
+            'fiscal_position_id': fiscal_position,
+            'partner_id': partner.id,
+        }
+
+    @mapping
+    def comment(self, record):
+        return {'comment': _('PrestaShop amount: %s') % record['amount']}
+
+    @mapping
+    @only_create
+    def invoice_lines(self, record):
+        slip_details = record.get(
+            'associations', {}
+        ).get('order_slip_details', []).get(
+            self.backend_record.get_version_ps_key('order_slip_detail'), [])
+        if isinstance(slip_details, dict):
+            slip_details = [slip_details]
+        lines = []
+        order_binding = self._get_order(record)
+        fpos = order_binding.fiscal_position_id
+        shipping_line = self._invoice_line_shipping(record, fpos)
+        if shipping_line:
+            lines.append((0, 0, shipping_line))
+        for slip_detail in slip_details:
+            line = self._invoice_line(slip_detail, fpos)
+            lines.append((0, 0, line))
+        return {'invoice_line_ids': lines}
+
+    def _invoice_line_shipping(self, record, fpos):
+        order_line = self._get_shipping_order_line(record)
```

Comment:
Hello there,
A customer requested the importation of Prestashop refunds and found a little problem:
Function self._get_shipping_order_line returns a list that includes one dictionary, but here, order_line is expected to be a dictionary.
This generate an error when importing Prestashop Refunds.

I confirm that Refund Importation works correctly when solving this issue.